### PR TITLE
[UNHOLY] Moth milk is here!

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

Okay so @Toastgoats made a moth milk sprite and I coded it. Whenever a non-moth drinks it, they get disgusted and if anything edible is splashed with it it will become moth-friendly food.

## Why It's Good For The Game

We need more memes and cum jokes

## Changelog
:cl: SmArtKar, ToastGoats
add: Added moth milk! It can be found in maints and will make any food cloth-like
/:cl:

### CODE REVIEW

**Comment by ArcaneMusic** [comment]
🤮

**Comment by Bytube** [comment]
but can we milk the moths

**Comment by Toastgoats** [comment]
"Yea, me milked moth. Sweet boy 'e woz."

**Comment by InsaneRed** [comment]
law 2 milk all moths on station

**Comment by SmArtKar** [comment]
We thought of moths "cumming" with that milk after lying in a bed with felinids but it was too much

**Comment by ArcaneDefence** [comment]
milkcarton
ship it

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 1]
```diff
@@ -1026,3 +1026,33 @@
 		M.drowsyness += 1 * REM * delta_time
 	return ..()
 
+
+/datum/reagent/consumable/moth_milk
+	name = "Moth Milk"
+	description = "Whoever thought that milking moths is a good idea was totally wrong. Is it even milk?"
```
Comment (ATH1909) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm:1032]:
```suggestion
	description = "Whoever thought that milking moths was a good idea was totally wrong. Is this even milk?"
```

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 2]
```diff
@@ -1026,3 +1026,33 @@
...
+	color = "#F0E9DA" // rgb: 240, 233, 218
+	taste_description = "salty and oily substance"
```
Comment (ATH1909) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm:1034]: oh god

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 3]
```diff
@@ -1026,3 +1026,33 @@
...
+	glass_name = "milk?"
+	glass_desc = "It's looking pretty odd, are you sure that it's safe to drink?"
```
Comment (ATH1909) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm:1036]:
```suggestion
	glass_desc = "It looks pretty odd, are you sure that it's safe to drink?"
```

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 4]
```diff
@@ -1026,3 +1026,33 @@
...
+			M.adjust_disgust(10 * REM * delta_time)
+		return TRUE
+	. = ..()
+
+/datum/reagent/consumable/moth_milk/expose_obj(obj/exposed_obj, reac_volume)
+	. = ..()
+	if(istype(exposed_obj, /obj/item/food))
+		var/obj/item/food/food = exposed_obj
+		food.foodtypes += CLOTH
+		food.visible_message("<span class='warning'>[food] sizzles and suddenly it's surface starts to look like it's made from some kind of... cloth?</span>")
```
Comment (ATH1909) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm:1052]:
```suggestion
		food.visible_message("<span class='warning'>[food] sizzles and suddenly its surface starts to look like it's made from some kind of... cloth?</span>")
```

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 5]
```diff
@@ -1026,3 +1026,33 @@
...
+		return
+
+	var/datum/component/edible/edible = exposed_obj.GetComponent(/datum/component/edible)
+	if(edible && istype(edible))
+		edible.foodtypes += CLOTH
+		exposed_obj.visible_message("<span class='warning'>[exposed_obj] sizzles and suddenly it's surface starts to look like it's made from some kind of... cloth?</span>")
```
Comment (ATH1909) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm:1058]:
```suggestion
		exposed_obj.visible_message("<span class='warning'>[exposed_obj] sizzles and suddenly its surface starts to look like it's made from some kind of... cloth?</span>")
```

**Comment by ATH1909** [comment]
mark this as an april fools pr please for the love of god

**Comment by SmArtKar** [comment]
> mark this as an april fools pr please for the love of god

who said that it was april fools, my friend?

**Comment by zombiegums** [comment]
cursed

**Comment by Mothblocks** [comment]
For the benefit of the maintainers, April Fools PRs will be labeled and closed. Key Holders are still able to test merge PRs that are closed, so this makes no difference to you, the author, assuming you were able to find a Key Holder to merge this PR in the first place.

**Comment by Serxule** [comment]
_YES YES YES YES YES YES_

**Comment by Serxule** [comment]
> add: Added moth milk! It can be found in maints and will make any food cloth-like

make it so we can harvest it from moths

**Comment by Mothblocks** [comment]
image

**Comment by Serxule** [comment]
we do a little trolling

**Comment by interestingusernam3** [comment]
If it makes food moth-edible...
Does this mean moths consume cum and all the clothing on the station is edible to them only because the washing machines cannot clean cum properly due to not using detergents?

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 6]
```diff
@@ -1026,3 +1026,33 @@
...
+	. = ..()
+	if(istype(exposed_obj, /obj/item/food))
+		var/obj/item/food/food = exposed_obj
+		food.foodtypes += CLOTH
+		food.visible_message("<span class='warning'>[food] sizzles and suddenly it's surface starts to look like it's made from some kind of... cloth?</span>")
+		return
+
+	var/datum/component/edible/edible = exposed_obj.GetComponent(/datum/component/edible)
+	if(edible && istype(edible))
+		edible.foodtypes += CLOTH
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm:1057]: this isnt how u use bitfields noob

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm (hunk 1 above)
Comment (Mothblocks) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm:1032]: ath went out of his way to review the moth fetish pr

Comment (ATH1909) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm:1032]: > ath went out of his way to review the moth fetish pr

https://www.youtube.com/watch?v=07wuZ3EWMAA

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm (hunk 6 above)
Comment (ATH1909) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm:1057]: nice catch, I missed that

you need to use |=, not +=

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm (hunk 2 above)
Comment (SmArtKar) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm:1034]: YES YES YES YES

**Comment by SmArtKar** [comment]
I didn't even say that it's a god damn april fools PR

**Comment by Mothblocks** [comment]
if its not april fools its not getting merged anyway

**Comment by The-Sun-In-Splendour** [comment]
image
image

**Comment by interestingusernam3** [comment]
consume the cum chalice

**Pull request closed without merging**